src/main/scala/ai/privado/tagger/sink/SinkArgumentUtility.scala

```diff
@@ -0,0 +1,333 @@
+    */
+  private def getLeftKey(
+    leftKey: AstNode,
+    topLeftKey: Option[String],
+    cpg: Cpg,
+    keyValueStructures: ListBuffer[(String, String)]
+  ): Option[String] = {
+    val res = Some(leftKey)
+      .collect {
+        case c: Call    => handleCallNode(c, topLeftKey, cpg, keyValueStructures).astChildren.isFieldIdentifier.head
+        case l: Literal => l
+      }
+      .map(_.code)
+    if (topLeftKey.getOrElse("").nonEmpty)
+      Some(topLeftKey.getOrElse("") + "." + res.getOrElse(""))
+    else
+      res
+  }
+
+  /** Extracts the left and right child nodes of a key-value pair. This function has generic logic to process all kinds
+    * of possible nodes { a: "test" | test | test() | x.test | { ... } }
+    * @return
+    *   A tuple containing the left and right child nodes as options.
+    */
+  private def extractKeyValPair(
+    objKeyVal: List[AstNode],
+    topLeftKey: Option[String],
+    cpg: Cpg,
+    keyValueStructures: ListBuffer[(String, String)]
+  ): (Option[String], Option[String]) = {
+    objKeyVal match {
+      case leftKey :: rightVal :: _ =>
+        (
+          getLeftKey(leftKey, topLeftKey, cpg, keyValueStructures),
+          Some(rightVal).collect {
+            case l: Literal         => l.code
+            case l: Identifier      => l.code
+            case l: FieldIdentifier => l.code
+            case l: Local           => l.code
+            case l: Call            => handleCallNode(l, topLeftKey, cpg, keyValueStructures).code
+            case l: Block =>
+              handleBlockNode(l, getLeftKey(leftKey, topLeftKey, cpg, keyValueStructures), cpg, keyValueStructures)
+          }
+        )
+      case _ =>
+        (None, None)
+    }
+  }
+
+  /** For payload sink node, maps it with available googlePayload event in obj Identifies the eventName & tries to map
+    * with blockNode key If matched further process the implemented method against it.
+    */
+  private def handlePayloadCallNode(
+    callNode: Call,
+    topLeftKey: String,
+    cpg: Cpg,
+    keyValueStructures: ListBuffer[(String, String)]
+  ): Unit = {
+    if (callNode.name.equals("payload")) {
+      val gpEventKey =
+        callNode.astChildren.isCall.astChildren.isCall.astChildren.isCall.astChildren.isFieldIdentifier.code.headOption
+          .getOrElse("")
+      val blockNode = cpg.call.code(".*tmp.*" + gpEventKey + " =.*").astChildren.isBlock.l
+
+      if (blockNode.nonEmpty) {
+        val internalBlockNode = blockNode.head.astChildren.isCall
+          .code(".*payload.*")
+          .astChildren
+          .isMethodRef
+          .referencedMethod
+          .astChildren
+          .isBlock
+          .l
+        val returnBlockNode = internalBlockNode.astChildren.isReturn.astChildren.isBlock.l
+        handleBlockNode(blockNode.head, Some(topLeftKey), cpg, keyValueStructures)
+        handleBlockNode(returnBlockNode.head, Some(topLeftKey), cpg, keyValueStructures)
+      }
+    }
+  }
```

Comment:
There are lot of breaking changes in this code, like usage of head. Add exception handling